File: TaskManagementProject/task-management-backend/src/main/java/com/example/taskmanagementbackend/models/Project.java
package com.example.taskmanagementbackend.models;

import com.example.taskmanagementbackend.models.enums.Category;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "projects")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;
    private String description;


    @Enumerated(EnumType.STRING)
    private Category category;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "project", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Task> tasks = new ArrayList<>();

    public Project(String name, String description, Category category, User user) {
        this.name = name;
        this.description = description;
        this.category = category;
        this.user = user;
    }

    // Getters and setters
}

File: TaskManagementProject/task-management-frontend/src/app/models/project.ts
import { Task } from "./task";

export enum Category{
  PROGRAMMING = 0,
  WORKOUT = 1,
  BASIC = 2
}
export interface Project {
  id: number;
  name: string;
  description: string;
  category: Category;
  userId: number;
  tasks: Task[];
}

export interface CreateProjectRequest {
  name: string;
  description: string;
  category: Category;
}

